core/domain/feedback_services.py

```diff
@@ -1182,3 +1188,64 @@ def delete_exploration_feedback_analytics(exp_ids):
         if model is not None]
     feedback_models.FeedbackAnalyticsModel.delete_multi(
         feedback_analytics_models_to_be_deleted)
+
+
+def handle_new_thread_created(exp_id):
+    """Reacts to new threads added to an exploration.
+
+    Args:
+        exp_id: str. The exploration ID associated with the thread.
+    """
+    _increment_total_threads_count_transactional(exp_id)
+    _increment_open_threads_count_transactional(exp_id)
+
+
+def handle_thread_status_changed(exp_id, old_status, new_status):
+    """Reacts to changes in an exploration thread's status.
+
+    Args:
+        exp_id: str. The exploration ID associated with the thread.
+        old_status: str. The old status of the thread.
+        new_status: str. The updated status of the thread.
+    """
+    # Status changed from closed to open.
+    if (old_status != feedback_models.STATUS_CHOICES_OPEN and
+            new_status == feedback_models.STATUS_CHOICES_OPEN):
+        _increment_open_threads_count_transactional(exp_id)
+    # Status changed from open to closed.
+    elif (old_status == feedback_models.STATUS_CHOICES_OPEN and
+          new_status != feedback_models.STATUS_CHOICES_OPEN):
+        _decrement_open_threads_count_transactional(exp_id)
+
+
+@transaction_services.run_in_transaction_wrapper
+def _increment_open_threads_count_transactional(exp_id):
+    """Increments count of open threads by one."""
+    model = (
+        feedback_models.FeedbackAnalyticsModel.get(exp_id, strict=False) or
+        feedback_models.FeedbackAnalyticsModel(id=exp_id, num_open_threads=0))
+    model.num_open_threads = (model.num_open_threads or 0) + 1
```

Comment:
Slightly odd to me that you need the default here -- in which case do you need to use it? If you retrieve a model from the datastore, all fields should already be there ... right?

Ditto below.

Reply:
Unfortunately not, since for these fields their default values are None.